app/src/main/java/net/aiscope/gdd_app/ui/capture/CaptureImageActivity.kt

```diff
@@ -34,6 +38,30 @@ class CaptureImageActivity : AppCompatActivity(), CaptureImageView {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_capture_image)
         setSupportActionBar(toolbar)
+        toolbar.setNavigationOnClickListener {
+            val alertDialog: AlertDialog? = this.let {
+                val builder = AlertDialog.Builder(it, R.style.MyAlertDialogStyle)
+                builder.apply {
+                    setPositiveButton(R.string.ok
+                    ) { _, _ ->
+                        // User clicked OK button
+                        val intent = Intent(context,  MainActivity::class.java)
+                        startActivity(intent)
+                    }
+                    setNegativeButton(R.string.cancel
+                    ) { dialog, _ ->
+                        // User cancelled the dialog
+                        dialog.dismiss()
+                    }
+                }
+                // Set other dialog properties
+                builder.setMessage("Are you sure you want to exit? Your current data will be lost")
+                builder.setTitle("Warning!")
+                // Create the AlertDialog
+                builder.create()
+            }
+            alertDialog!!.show()
```

Comment:
Is there a way not to use the double-bang operator here? Since it is considered an antipattern usually.